source/MRMesh/MRMeshBoolean.cpp

```diff
@@ -230,6 +260,71 @@ BooleanResult boolean( const Mesh& meshA, const Mesh& meshB, BooleanOperation op
     return result;
 }
 
+BooleanResultPoints getIntersectionAndInnerPoints( const Mesh& meshA, const Mesh& meshB, const AffineXf3f* rigidB2A )
+{
+    MR_TIMER
+
+    BooleanResultPoints result;
+    result.meshAVerts.resize( meshA.topology.lastValidVert() + 1 );
+    result.meshBVerts.resize( meshB.topology.lastValidVert() + 1 );
+
+    const auto convertersB2A = getVectorConverters( meshA, meshB, rigidB2A );
+
+    const auto xfB2A = rigidB2A ? rigidB2A->inverse() : AffineXf3f{};
+    const auto* rigidA2B = rigidB2A ? &xfB2A : nullptr;
+    const auto convertersA2B = getVectorConverters( meshB, meshA, rigidB2A );
+
+    const auto intersections = findCollidingEdgeTrisPrecise( meshA, meshB, convertersB2A.toInt, rigidB2A );
+    result.intersectionPoints.reserve( intersections.edgesAtrisB.size() + intersections.edgesBtrisA.size() );
+
+    FaceBitSet collFacesA, collFacesB;
+    VertBitSet collOuterVertsA, collOuterVertsB;
+    collFacesA.resize( meshA.topology.lastValidFace() + 1 );
+    collFacesB.resize( meshB.topology.lastValidFace() + 1 );
+    collOuterVertsA.resize( meshA.topology.lastValidVert() + 1 );
+    collOuterVertsB.resize( meshB.topology.lastValidVert() + 1 );
+    for ( const auto& et : intersections.edgesAtrisB )
+    {
+        collFacesA.set( meshA.topology.left( et.edge ) );
+        collFacesA.set( meshA.topology.right( et.edge ) );
+        collFacesB.set( et.tri );
+        collOuterVertsA.set( meshA.topology.dest( et.edge ) );
+
+        const auto isect = findEdgeTriIntersectionPoint( meshB, et.tri, meshA, et.edge, rigidA2B, &convertersA2B );
+        result.intersectionPoints.emplace_back( rigidB2A ? ( *rigidB2A )( isect ) : isect );
+    }
+    for ( const auto& et : intersections.edgesBtrisA )
+    {
+        collFacesB.set( meshB.topology.left( et.edge ) );
+        collFacesB.set( meshB.topology.right( et.edge ) );
+        collFacesA.set( et.tri );
+        collOuterVertsB.set( meshB.topology.dest( et.edge ) );
+
+        const auto isect = findEdgeTriIntersectionPoint( meshA, et.tri, meshB, et.edge, rigidB2A, &convertersB2A );
+        result.intersectionPoints.emplace_back( isect );
+    }
```

Comment:
better use only one rigidB2a and one converterB2A, just apply it to B mesh points only

Reply:
this blocks have more similar code